tests/functional/tests/cou_smoke.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+import unittest
+from subprocess import check_call, check_output
+
+import zaza
+
+log = logging.getLogger(__name__)
+
+
+class COUSmokeTest(unittest.TestCase):
+    """COU smoke functional tests."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.model_name = zaza.model.get_juju_model()
+        cls.install_package()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.remove_package()
+
+    def install_package() -> None:
+        """Install cou package."""
+        cou_test_snap = os.environ.get("TEST_SNAP")
+        if cou_test_snap:
+            log.info(f"Installing {cou_test_snap}")
+            assert os.path.isfile(cou_test_snap)
+            # install the snap
+            assert check_call(f"sudo snap install --dangerous {cou_test_snap}".split()) == 0
+            # connect interfaces
+            interfaces_to_connect = [
+                "juju-client-observe",
+                "dot-local-share-cou",
+                "ssh-public-keys",
+            ]
+            for interface in interfaces_to_connect:
+                assert (
+                    check_call(
+                        f"sudo snap connect charmed-openstack-upgrader:{interface} snapd".split()
+                    )
+                    == 0
+                )
+            # make the cou alias
+            assert check_call("sudo snap alias charmed-openstack-upgrader.cou cou".split()) == 0
+
+            assert check_output("which cou".split()).decode().strip() == os.path.join(
+                "/snap/bin/cou"
+            )
+        else:
+            log.warning("Installing python package")
+            assert check_call("python3 -m pip install .".split()) == 0
+            assert (
+                check_output("which cou".split())
+                .decode()
+                .strip()
+                .startswith(os.path.join(os.getcwd(), ".tox"))
+            )
+
+    def remove_package() -> None:
+        """Remove cou package."""
+        cou_test_snap = os.environ.get("TEST_SNAP")
+        if cou_test_snap:
+            log.info("Removing snap package cou")
+            check_call("sudo snap remove charmed-openstack-upgrader --purge".split())
+        else:
+            logging.info("Uninstalling python package cou")
+            check_call("python3 -m pip uninstall --yes charmed-openstack-upgrader".split())
+
+    def test_plan_backup(self) -> None:
+        """Test plan with backup."""
+        result = check_output("cou plan".split()).decode()
```

Comment:
We should not call `cou` direcly, but rather through helper function. e.g, `cou("plan", "--backup")` or `cou_run("plan")`